server/src/main/java/io/druid/initialization/Initialization.java

```diff
@@ -142,6 +142,8 @@ static void clearLoadedModules()
               "Extension module [%s] was ignored because it doesn't have a canonical name, is it a local or anonymous class?",
               module.getClass().getName()
           );
+        } else if (config.getExcludeModules().contains(moduleName)) {
```

Comment:
I don't want to do this for a number of reasons:
 - First of all, we don't need it
 - It's more complex and raises small issues on implementation level, e. g. for regex you need to recognize package dots which are not regex dots
 - List of class names is the type of property which is already used in Druid, don't want to add more types like regexes, globs patterns, etc.
 - If somebody want to exclude many modules from some extension, likely it's a better idea to extract those modules into a separate extension and simply not add this new extension to the loadList.